### PR TITLE
Device: Set update lock

### PR DESCRIPTION
**Pull request by karaxuna**
Connects-to: #635
Change-type: minor

### CODE REVIEW

File: lib/models/device.coffee [hunk 1]
```diff
@@ -1760,6 +1760,70 @@ getDeviceModel = (deps, opts) ->
...
+	#
+	# @example
+	# balena.models.device.setLockOverriden(123, true);
+	#
+	# @example
+	# balena.models.device.setLockOverriden('7cf02a6', true, function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.setLockOverriden = (uuidOrId, shouldOverride, callback) ->
```
Comment (sradevski) [inline lib/models/device.coffee]: I personally find methods like `enableLockOverride` and `disableLockOverride` more intuitive to use and more difficult to call with the wrong argument, so maybe we can name it like that? The `setLockOverriden` one can be private and stay pretty much the same (except maybe pass `'1'` and `'0'`) and be called by the two methods I mentioned above.

File: lib/models/device.coffee [hunk 2]
```diff
@@ -1760,6 +1760,96 @@ getDeviceModel = (deps, opts) ->
...
+	#
+	# @example
+	# balena.models.device.isLockOverriden(123);
+	#
+	# @example
+	# balena.models.device.isLockOverriden('7cf02a6', function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.isLockOverriden = (uuidOrId, callback) ->
```
Comment (thgreasi) [inline lib/models/device.coffee]: How about naming this as `hasLockOverride` similarly to `hasDeviceUrl`?

File: typings/balena-sdk.d.ts [hunk 3]
```diff
@@ -839,6 +845,9 @@ declare namespace BalenaSdk {
 					expiryTimestamp: number,
 				): Promise<void>;
 				revokeSupportAccess(uuidOrId: string | number): Promise<void>;
+				enableLockOverride(uuidOrId: string | number): Promise<void>;
+				disableLockOverride(uuidOrId: string | number): Promise<void>;
+				isLockOverriden(uuidOrId: string | number): Promise<boolean>;
```
Comment (thgreasi) [inline typings/balena-sdk.d.ts]: Reminder
> How about naming this as `hasLockOverride` similarly to `hasDeviceUrl`?

**Comment by thgreasi** [comment]
Let's also title the commit with something like `Add device lock override methods`.

**Comment by thgreasi** [comment]
The tests for the new methods do pass but the test suit times out on both travis & appveyor.
Merging as an admin.

**Pull request merged**